herd/AArch64Sem.ml

```diff
@@ -1991,9 +1991,11 @@ module Make
         | M.A.V.Val (Concrete i) -> Some (B.Addr (M.A.V.Cst.Scalar.to_int i))
         | _ -> None
 
-      let do_indirect_jump test bds i v =
+      let do_indirect_jump test bds i ii v =
         match  v2tgt v with
-        | Some tgt -> M.unitT (B.Jump (tgt,bds))
+        | Some tgt ->
+          commit_bcc ii
+          >>= fun () -> M.unitT (B.Jump (tgt,bds))
```

Comment:
In my opinion `commit_bcc ii >>= fun () ->` has to be added in front of line  2008 below.